src/compiler/checker.ts

```diff
@@ -42086,12 +42100,8 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         // Check if the index type is assignable to 'keyof T' for the object type.
         const objectType = (type as IndexedAccessType).objectType;
         const indexType = (type as IndexedAccessType).indexType;
-        // skip index type deferral on remapping mapped types
-        const objectIndexType = isGenericMappedType(objectType) && getMappedTypeNameTypeKind(objectType) === MappedTypeNameTypeKind.Remapping
-            ? getIndexTypeForMappedType(objectType, IndexFlags.None)
-            : getIndexType(objectType, IndexFlags.None);

```

Comment:
This reverts my own change from https://github.com/microsoft/TypeScript/pull/55140 . I think now the check wasn't exhaustive anyway and this is now better handled by `getSimplifiedIndexType`